docs: add license information (#8164)
Homebrew/brew: docs/Formula-Cookbook.md

Try to summarise from the [`homepage`](https://rubydoc.brew.sh/Formula#homepage%3D-class_method) what the formula does in the [`desc`](https://rubydoc.brew.sh/Formula#desc%3D-class_method)ription. Note that the [`desc`](https://rubydoc.brew.sh/Formula#desc%3D-class_method)ription is automatically prepended with the formula name.

### Fill in the `license`

> **We don’t accept formulae without a [`license`](https://rubydoc.brew.sh/Formula#license-class_method)!**

Reviewer: ```suggestion
**We don’t accept new formulae into Homebrew/homebrew-core without a [`license`](https://rubydoc.brew.sh/Formula#license-class_method)!**
```

> Find the license identifier from the [SPDX License List](https://spdx.org/licenses/). We only accept licenses from this list.

Reviewer: ```suggestion
Find the license identifier from the [SPDX License List](https://spdx.org/licenses/). We only accept licenses from this list that are a [Debian Free Software Guidelines license](https://wiki.debian.org/DFSGLicenses).
```

> Find the license identifier from the [SPDX License List](https://spdx.org/licenses/). Note that we don't accept licenses that don't appear on the list.

Reviewer 2: ```suggestion
Find the license identifier from the [SPDX License List](https://spdx.org/licenses/). We only accept licenses from this list.
```

> If the formula gives the user the option to choose which license to use, you should list them all in an array:

Reviewer: ```suggestion
If the software is available under multiple licenses, you should list them all in an array:
```

```ruby
license ["MIT", "GPL-2.0"]
```

> Note: only specify multiple licenses if the formula gives the user a choice between the licenses. Formulae that have different licenses for different parts of their software should specify only the more restrictive license.

Reviewer 3: How does one objectively determine which license is "more restrictive"?

Author: Umm... not sure...

Author: Is this soemthing that's been discussed elsewhere? If not, we should figure this out ASAP

Reviewer 3: I have no idea.

I think we need a mechanism in DSL for this so that we can supply a list of licenses and say whether it is a choice or whether they all apply, like an `any/all` second parameter or something. This would then in turn affect how `HOMEBREW_FORBIDDEN_LICENSES` behaves.

Reviewer: Yes, there is an objective way of determining this. I can't find a better source than this (although I'm sure I've seen some) but https://choosealicense.com/licenses/ is good and , for the longer tail, https://en.wikipedia.org/wiki/Comparison_of_free_and_open-source_software_licences has a table that generally can be used to compare permissiveness. It may not appear obvious from the table how to compare any two arbitrary licenses but it should be from choosealicense.com and for the examples we'll actually see in the wild.

Author: 👍 I added those resources as links in the text.
